src/api/contact-form.ts

```diff
@@ -0,0 +1,105 @@
+import { WebClient, LogLevel } from '@slack/web-api';
+import { GatsbyFunctionRequest, GatsbyFunctionResponse } from 'gatsby';
+
+const TOKEN = process.env.SLACK_BOT_SY_TOKEN;
+
+/**
+ * Easily build this block structure via https://app.slack.com/block-kit-builder
+ */
+const createSlackMessage = ({ name, email, message }) => {
+  return [
+    {
+      type: 'header',
+      text: {
+        type: 'plain_text',
+        text: 'New form submission',
+      },
+    },
+    {
+      type: 'section',
+      text: {
+        type: 'mrkdwn',
+        text: 'We have received a new message.',
+      },
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: `*Name*:\n ${name}`,
+        },
+        {
+          type: 'mrkdwn',
+          text: `*E-mail*:\n ${email}`,
+        },
+      ],
+    },
+    {
+      type: 'divider',
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'plain_text',
+          text: message,
+        },
+      ],
+    },
+    {
+      type: 'context',
+      elements: [
+        {
+          type: 'mrkdwn',
+          text: 'Sent via _Satellytes Contact Form_',
+        },
+      ],
+    },
+  ];
+};
+
+export default async function handler(
+  req: GatsbyFunctionRequest,
+  res: GatsbyFunctionResponse,
+) {
+  const { name, email, message, ...honey } = req.body;
+  console.log('receive form data', { name, email, message }, honey);
+
+  if (honey.firstName || honey.phone) {
+    console.log(`honeypot trap filled, aborting (received "${honey}")`);
+
+    // this is probably a bot as our honeypot fields have been filled
+    // let's silently ignore that bot
+    return res.status(200).json({ ok: true });
+  }
+
+  if (req.method !== `POST`) {
+    return res.status(500).json({ error: `not allowed` });
+  }
+
+  if (!TOKEN) {
+    return res.status(500).json({ error: `no slack token given` });
+  }
+
+  const channelId = 'website';
```

Comment:
We could also use the more anonymous channel id but I think we are not harming anyone here by using the plain name 🙏